src/comment.rs

```diff
@@ -81,11 +81,11 @@ pub fn rewrite_comment(orig: &str,
             let rewrite = try_opt!(rewrite_string(line, &fmt));
             result.push_str(&rewrite);
         } else {
-            if line.len() == 0 {
-                result.pop(); // Remove space if this is an empty comment.
-            } else {
-                result.push_str(line);
+            if line.len() == 0 || line.starts_with('!') {
```

Comment:
This strips the whitespace in between the `//` and `!`. Otherwise we'd format
```rust
//! comment
```
as
```rust
// ! comment
```
Not sure about about `///` comments. Those usually aren't usually caught in missing spans, but I'll check.